### PR TITLE
Mepts 363 rm changes in monthly report misau b13

### PR DESCRIPTION
**Pull request by cfaife**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java [hunk 1]
```diff
@@ -488,4 +488,22 @@ public CohortDefinition hasEncounter(EncounterType... types) {
...
     return cd;
   }
+  
+  /**
+   * Get patients who have encounter of a specific before endDate
+   *
+   * @param encounterType
+   * @retrun CohortDefinition
+   */
+  public CohortDefinition getPatientsHavingEncounterBeforeEndDate(int encounterType) {
```
Comment (simaomazive) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java]: @cfaife in adition to use` e.encounter_type = %d` can you update to `e.encounter_type = ${encounter_type}`

Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java]: thanks @simaomazive, I will fix that right now

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java]: Guys lets not use SQL for this simple queries. We should be using `EncounterCohortDefinition`.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/resumo/ResumoMensalDataSetDefinition.java [hunk 2]
```diff
@@ -363,6 +363,31 @@ public DataSetDefinition constructResumoMensalDataset() {
         getPatientsWhoWereActiveByEndOfPreviousMonth(),
         resumoMensalAandBdisaggregations.getAdolescentesColumns());
 
+    // B13 Indicators
+    dsd.addColumn(
+        "B13TP",
+        "Total patients - Total Geral",
+        this.getActivePatientsInARTByEndOfCurrentMonth(),
```
Comment (ynurmahomed) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/resumo/ResumoMensalDataSetDefinition.java]: We don't need to use `this`. I bet this is an eclipse auto thing.

Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/resumo/ResumoMensalDataSetDefinition.java]: ok, but  why we shouldn't  use  `this`?, it is not, I did this intentionally

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/resumo/ResumoMensalDataSetDefinition.java]: `This` is implicit, even if you don't add it, the method call will default to the class scope. Apart from that, all other code in the project omits `this` when not strictly necessary.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java (hunk 1 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java]: Makes sense  @ynurmahomed, but its  already changed

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java]: A ideia é não reinventar a roda. 

Esta query já é implementada pelo OpenMRS (`EncounterCohortDefinition`). É só uma questão de instanciar a classe e dar os parâmetros certos:

https://github.com/esaude/openmrs-module-eptsreports/blob/d5cb03313f87307c023d0b1d390e369a32736740/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java#L474-L490

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 3]
```diff
@@ -1110,4 +1110,61 @@ public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthFor
...
+    CohortDefinition B8E = getPatientsWhoDied(false);
+
+    String mappings = "onOrAfter=${startDate},onOrBefore=${endDate},location=${location}";
+
+    String mappingsOnDate = "onDate=${endDate},location=${location}";
+    String mappingsOnOrBeforeLocationList = "onOrBefore=${endDate},locationList=${location}";
+
+    cd.addSearch("startedArt", map(startedArt, mappings));
+
+    cd.addSearch("fila", map(fila, "nOrBefore=${endDate},locationList=${location}"));
```
Comment (amosh101) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: O parametro da data precisa de atencao, deveria ser "onOrBefore"?

**Review by amosh101 — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java (hunk 3 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: fixed

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 4]
```diff
@@ -1192,4 +1192,61 @@ public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthFor
...
+
+    cd.addSearch("startedArt", map(startedArt, mappings));
+
+    cd.addSearch("fila", map(fila, "onOrBefore=${endDate},locationList=${location}"));
+
+    cd.addSearch("masterCardPickup", map(masterCardPickup, mappingsOnOrBeforeLocationList));
+
+    cd.addSearch("transferredIn", map(transferredIn, mappings));
+    cd.addSearch("B5E", map(B5E, mappings));
+    cd.addSearch("B6E", map(B6E, "onOrBefore=${endDate},locationList=${location}"));
```
Comment (amosh101) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: The parameter locationList needs to be updated for location, used in the new B6 query.

**Review by amosh101 — CHANGES_REQUESTED** [review]
Needs a small update.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 5]
```diff
@@ -1192,4 +1192,61 @@ public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthFor
...
+
+    CohortDefinition masterCardPickup = getPatientsWithMasterCardDrugPickUpDate();
+
+    CohortDefinition transferredIn =
+        getNumberOfPatientsTransferredInFromOtherHealthFacilitiesDuringCurrentMonthB2();
+
+    CohortDefinition B5E = getPatientsTransferredOutB5();
+
+    CohortDefinition B6E = getPatientsWhoSuspendedTreatmentB6();
+    ;
```
Comment (amosh101) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: We can delete this comma.

Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java (hunk 4 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 6]
```diff
@@ -1192,4 +1192,60 @@ public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthFor
...
+
+    cd.addSearch("startedArt", map(startedArt, mappings));
+
+    cd.addSearch("fila", map(fila, "onOrBefore=${endDate},locationList=${location}"));
+
+    cd.addSearch("masterCardPickup", map(masterCardPickup, mappingsOnOrBeforeLocationList));
+
+    cd.addSearch("transferredIn", map(transferredIn, mappings));
+    cd.addSearch("B5E", map(B5E, mappings));
+    cd.addSearch("B6E", map(B6E, "onOrBefore=${endDate},location=${location}"));
```
Comment (amosh101) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java:1347]: On B12 requirement we need cumulative number of active patients in ART by end of current period, B6 currently uses two dates as parameters for an interval period, (between onOrAfter and onOrBefore), B6 query needs an update to display all ART patients until the end of the period, (<= onOrBefore).

**Review by amosh101 — CHANGES_REQUESTED** [review]

**Comment by ningosi** [comment]
For me to finalize on section E of this report, I need this ticket to be finalized and merged into develop.

**Pull request merged**